### PR TITLE
[iptables] Add mapping for journald.custom fields

### LINKED ISSUE #11337: [Stack 8.16.0-SNAPSHOT] [iptables] Failing test daily: system test: journald in iptables.log
- Stack version: 8.16.0-SNAPSHOT
- Package: iptables
- Failing test: system test: journald
- DataStream: log
- Owners:
    - @elastic/sec-deployment-and-devices

Failure:
```
test case failed: one or more errors found in documents stored in logs-iptables.log-83342 data stream: [0] field "journald.custom.seqnum" is undefined
[1] field "journald.custom.seqnum_id" is undefined
```

Latest issue closed for the same test: https://github.com/elastic/integrations/issues/10950

First build failed: https://buildkite.com/elastic/integrations/builds/16819

Latest 5 failed builds:
- https://buildkite.com/elastic/integrations/builds/17520
- https://buildkite.com/elastic/integrations/builds/17551
- https://buildkite.com/elastic/integrations/builds/17585
- https://buildkite.com/elastic/integrations/builds/17590
- https://buildkite.com/elastic/integrations/builds/17594

### LINKED ISSUE #11559: [Stack 8.17.0-SNAPSHOT] [iptables] Failing test daily: system test: journald in iptables.log
- Stack version: 8.17.0-SNAPSHOT
- Package: iptables
- Failing test: system test: journald
- DataStream: log
- Owners:
    - @elastic/sec-deployment-and-devices

Failure:
```
test case failed: one or more errors found in documents stored in logs-iptables.log-95566 data stream: [0] field "journald.custom.seqnum" is undefined
[1] field "journald.custom.seqnum_id" is undefined
```

First build failed: https://buildkite.com/elastic/integrations/builds/17669

### PR DESCRIPTION
**Pull request by mjwolf**
Fixes this error in system tests:
```
test case failed: one or more errors found in documents stored in logs-iptables.log-95566 data stream: [0] field "journald.custom.seqnum" is undefined
[1] field "journald.custom.seqnum_id" is undefined
```

e.g. https://buildkite.com/elastic/integrations/builds/17669

## Proposed commit message

Add mappings for `journald.custom` fields, which will fix an automated system test failure.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] ~~I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices)~~

## How to test this PR locally

Run `elastic-package test system`, there should be no failures

## Related issues

- Closes #11337 
- Closes #11559

### CODE REVIEW

**Comment by andrewkroh** [comment]
SEQNUM_ID is not a "custom" field. This is built-in to journald. IMO this is an input bug that it writes it to the "journald.custom" namespace. We should fix that problem.

Could we ship an update to the ingest pipeline that drops this specific field instead of using the flattened? The field is not particularly useful IMO and takes up a lot of space.

- https://www.freedesktop.org/software/systemd/man/latest/systemd.journal-fields.html#Address%20Fields

**Comment by mjwolf** [comment]
> SEQNUM_ID is not a "custom" field. This is built-in to journald. IMO this is an input bug that it writes it to the "journald.custom" namespace. We should fix that problem.
> 
> To address the test failure, could we ship an update to the ingest pipeline that drops this specific field instead of using the flattened? The field is not particularly useful IMO and takes up a lot of space.
> 
> * https://www.freedesktop.org/software/systemd/man/latest/systemd.journal-fields.html#Address%20Fields

Thanks, that's good to know. I've removed the seqnum fields in the ingest pipeline

File: packages/iptables/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,10 @@
 # newer versions go on top
+- version: "1.17.0"
+  changes:
+    - description: Remove journald seqnum fields
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/999999
+
```
Comment (andrewkroh) [inline packages/iptables/changelog.yml:7]:
```suggestion
      link: https://github.com/elastic/integrations/pull/11839
```

File: packages/iptables/changelog.yml [hunk 2]
```diff
@@ -1,4 +1,10 @@
 # newer versions go on top
+- version: "1.17.0"
+  changes:
+    - description: Remove journald seqnum fields
```
Comment (andrewkroh) [inline packages/iptables/changelog.yml:4]:
```suggestion
    - description: Remove journald.custom.seqnum and journald.custom.seqnum_id which are misclassified as custom fields by the journald input.
```

**Review by andrewkroh — APPROVED** [review]
LGTM. Will you please open an elastic/beats issue to track the underlying input problem that neither field should be sent through as "custom".

**Comment by andrewkroh** [comment]
@mjwolf Sorry, I didn't notice that auto-merge was enabled. I did have a suggestion and one request.

**Pull request merged**